### PR TITLE
Backend: Notifications

### PR DESCRIPTION
**Pull request by kave**
*Note* Integration into Filterable List will happen in a Future PR

screen shot 2016-01-04 at 11 53 02 am
@anselmbradford 
@kurtw

### CODE REVIEW

File: cfgov/v1/models/molecules.py [hunk 1]
```diff
@@ -226,3 +226,20 @@ class Meta:
...
+                                                         "depending on notification type.")
+    explanation = blocks.TextBlock(required=False,
+                                   help_text="Text to display as an explanation within the notification markup.")
+
+    class Meta:
+        icon = 'view'
+        template = '_includes/molecules/notification.html'
+
+    class Media:
+        js = ("notification.js",)
```
Comment (kurtw) [inline cfgov/v1/models/molecules.py]: God I love this.

Comment (kave) [inline cfgov/v1/models/molecules.py]: :dancer:

**Comment by kurtw** [comment]
I'm a bit unclear on how we're going to be using this. With this, we have a way of creating 3 different types of notifications but how do we identify certain `error` or `warning` notifications for different situations?

Would it make sense to have a notification on a "per field" basis? e.g. The date range has it's own set of notifications.

**Comment by kave** [comment]
@kurtw take a look at this PR https://github.com/cfpb/cfgov-refresh/pull/1274 pretty much these would be used in conjunction with other organisms (probably as streamBlocks within them) if you wanted to customize the messages presented by the notifications.

File: cfgov/unprocessed/js/routes/on-demand/expandable.js [hunk 2]
```diff
@@ -18,12 +17,4 @@ var expandable;
...
-}
-
-selector = '.o-expandable-group';
-var expandableGroup = document.querySelectorAll( selector );
-var regularExpandableGroup = new ExpandableGroup( expandableGroup[0] );
-var accordionExpandableGroup = new ExpandableGroup( expandableGroup[1] );
-
-regularExpandableGroup.init();
-accordionExpandableGroup.init();
+}
```
Comment (jimmynotjim) [inline cfgov/unprocessed/js/routes/on-demand/expandable.js]: Minor, add newline at end of file back in

File: cfgov/unprocessed/js/routes/on-demand/notification.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+   ========================================================================== */
+
+'use strict';
+
+var Notification = require( '../../molecules/Notification' );
+
+var notifications = document.querySelectorAll( '.m-notification' );
+var notification;
+for ( var i = 0, len = notifications.length; i < len; i++ ) {
+	notification = new Notification( notifications[i] ).init();
```
Comment (jimmynotjim) [inline cfgov/unprocessed/js/routes/on-demand/notification.js]: Extra indentation, JS is 2-space

**Comment by kurtw** [comment]
Right, but that doesn't address when specific messages are used. I'm assuming that we want specific error and warning messages to be shown in a scenario, like when the user inputs the wrong thing or something.

I think what I'm saying is the we should have an identifying field for a notification so we know which one we're showing.

**Comment by kave** [comment]
@kurtw eh, i mean sure but do we really want to have big notifications for on field errors? We can just use form errors to highlight incorrect fields. i think these notifications are primarily for submission notifications @anselmbradford  what do you think?

**Comment by anselmbradford** [comment]
If I recall correctly, the field errors would show in cases where a browser supports them, but not all browsers do, so the main notification would need to be capable of displaying them also in those cases.

**Comment by kave** [comment]
@anselmbradford really? thats a first. now i'm just curious for my own knowledge what browsers did you see this happen? IE ?

**Comment by anselmbradford** [comment]
@kave Yeah less than IE10 and Safari (surprisingly)—which supports the form validation attributes but doesn't display a notice. http://caniuse.com/#search=required

**Comment by kave** [comment]
@kurtw so after speaking it over with ans, we decided the most consistent approach is to just have the bulleted list of forms errors added to the notification `explanation` section.

having a `field name` box may over complicate things as content would have to know the id/name of the html fields their associating the notifications too and these would potentially be auto-generated.

**Pull request merged**